### PR TITLE
Add new subpackage cirq-ft

### LINKED ISSUE #6119: Create a new sub-package for `cirq-ft`
We want to launch Cirq-FT soon. This task is to create a new empty sub-package for `cirq-ft` that depends on `cirq-core` and is compatible with all the Cirq packaging infrastructure. 

Once the package has been added, we can start moving files from internal private repository to Cirq open source.

### PR DESCRIPTION
**Pull request by pavoljuhas**
- add empty bare bones cirq-ft package
- address pylint recommendation about opening files in setup.py
- include cirq-ft/requirements.txt in cirq-all-no-contrib.txt

Partially fixes #6119

### CODE REVIEW

File: cirq-ft/README.rst [hunk 1]
```diff
@@ -0,0 +1,21 @@
+.. image:: https://raw.githubusercontent.com/quantumlib/Cirq/master/docs/images/Cirq_logo_color.png
+  :target: https://github.com/quantumlib/cirq
+  :alt: cirq-ft
+  :width: 500px
+
+Cirq is a Python library for writing, manipulating, and optimizing quantum
+circuits and running them against quantum computers and simulators.
+
+This module is **cirq-ft**, which enhances support for fault-tolerant algorithms
+and provides functions for quantum memory management.
```
Comment (viathor) [inline cirq-ft/README.rst:10]: Cirq's github homepage defines it as

> A python framework for creating, editing, and invoking Noisy Intermediate Scale Quantum (NISQ) circuits.

Wikipedia says that

> Cirq is an open-source framework for noisy intermediate scale quantum (NISQ) computers.

Perhaps the README.md should discuss (and maybe even resolve!) this apparent contradiction.

File: cirq-ft/setup.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+from setuptools import find_packages, setup
+
+# This reads the __version__ variable from cirq/_version.py
+__version__ = ''
+exec(pathlib.Path('cirq_ft/_version.py').read_text(encoding='utf-8'))
+
+name = 'cirq-ft'
+
+description = 'A Cirq package to fault-tolerant algorithms'
```
Comment (viathor) [inline cirq-ft/setup.py:26]: s/to/for/

File: cirq-ft/README.rst (hunk 1 above)
Comment (pavoljuhas) [reply cirq-ft/README.rst:12]: Ack.  This PR is to add a minimum package for transferring code from a development non-public repository.
The README file is not yet ready; I have added a TODO marker it needs to be reviewed.

File: cirq-ft/setup.py (hunk 2 above)
Comment (pavoljuhas) [reply cirq-ft/setup.py]: Thank you, fixed.

File: cirq-ft/cirq_ft/_version.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Define version number here, read it from setup.py automatically"""
+
+__version__ = "1.2.0.dev"
```
Comment (tanujkhattar) [inline cirq-ft/cirq_ft/_version.py:17]: @verult Can we do `0.1.0dev` instead?

**Review by tanujkhattar — CHANGES_REQUESTED** [review]

File: cirq-ft/cirq_ft/_version.py (hunk 3 above)
Comment (verult) [reply cirq-ft/cirq_ft/_version.py:17]: A big benefit of keeping the same version is that the compatibility matrix between vendor packages and cirq-core is simple: vendor package version 1.2.3 depends on cirq-core 1.2.3. I think using 0.x.x for a new packages makes sense, and we can even bump major and minor versions at a different cadence as cirq-core if our release tools allow, but we should then document compatibility with cirq-core.

For example, if say cirq-ft 0.3.0 requires cirq-core 1.3 features:

* cirq-ft 0.1 - 0.2 : >= cirq-core 1.2
* cirq-ft 0.3: >= cirq-core 1.3

Comment (pavoljuhas) [reply cirq-ft/cirq_ft/_version.py:17]: IIUC the plan is to distribute cirq-ft together with cirq.  For example, the cirq metapackage currently includes all submodules including cirq-ft.

https://github.com/quantumlib/Cirq/blob/49b2cfb0a7568e4fd7e92cc563c8291a2a88b645/setup.py#L53-L54

If that is the case, I think it is better to keep cirq-ft version in sync with all other versions in cirq.  
We can consider cirq-ft to be a new capability provided with the next public version Cirq-1.2.0.

**Review by tanujkhattar — APPROVED** [review]
This is exciting! Thanks Pavol!

I do have a slight discomfort in launching a new sub-package directly at version 1.2; but as discussed, this is a small price to pay for integrating well with rest of the cirq ecosystem!

**Pull request merged**